pkg/network/protocols/http/kafka_monitor_test.go

```diff
@@ -0,0 +1,393 @@
+	"github.com/DataDog/datadog-agent/pkg/network/protocols/http/testutil"
+	"github.com/DataDog/datadog-agent/pkg/network/protocols/kafka"
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+
+	"github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/kversion"
+)
+
+var defaultTopicName = "franz-kafka"
+
+type BinaryType int
+
+const (
+	PREBUILT = 0
+	RUNTIME  = 1
+	CORE     = 2
+)
+
+// This test loads the Kafka binary, produce and fetch kafka messages and verifies that we capture them
+func TestSanity(t *testing.T) {
+	skipTestIfKernelNotSupported(t)
+	kafka.RunServer(t, "127.0.0.1", "9092")
+	monitor := newHTTPWithKafkaMonitor(t)
+
+	seeds := []string{"localhost:9092"}
+	client, err := kgo.NewClient(
+		kgo.SeedBrokers(seeds...),
+		kgo.DefaultProduceTopic(defaultTopicName),
+		kgo.ConsumeTopics(defaultTopicName),
+		kgo.MaxVersions(kversion.V2_5_0()),
+	)
+	require.NoError(t, err)
+	defer client.Close()
+	ctxTimeout, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	err = client.Ping(ctxTimeout)
+	cancel()
+	require.NoError(t, err)
+
+	// Create the topic
+	adminClient := kadm.NewClient(client)
+	ctxTimeout, cancel = context.WithTimeout(context.Background(), time.Second*5)
+	_, err = adminClient.CreateTopics(ctxTimeout, 1, 1, nil, defaultTopicName)
+	cancel()
+	require.NoError(t, err)
+
+	record := &kgo.Record{Topic: defaultTopicName, Value: []byte("Hello Kafka!")}
+	ctxTimeout, cancel = context.WithTimeout(context.Background(), time.Second*5)
+	err = client.ProduceSync(ctxTimeout, record).FirstErr()
+	cancel()
+	require.NoError(t, err, "record had a produce error while synchronously producing")
+
+	fetches := client.PollFetches(context.Background())
+	errs := fetches.Errors()
+	for _, err := range errs {
+		t.Errorf("PollFetches error: %+v", err)
+		t.FailNow()
+	}
+
+	// We expect 2 occurrences for each connection as we are working with a docker
+	expectedStatsCount := 4
+	statsCount := 0
+	var kafkaStats map[kafka.Key]*kafka.RequestStat
+	require.Eventually(t, func() bool {
+		kafkaStats = monitor.GetKafkaStats()
+		statsCount = len(kafkaStats)
+		return expectedStatsCount == statsCount
+	}, time.Second*3, time.Millisecond*100, "Expected to find a %d stats, instead captured %d", expectedStatsCount, statsCount)
+
+	numberOfProduceRequests := 0
+	numberOfFetchRequests := 0
+	for kafkaKey, kafkaStat := range kafkaStats {
+		require.Equal(t, defaultTopicName, kafkaKey.TopicName)
+		switch kafkaKey.RequestAPIKey {
+		case kafka.ProduceAPIKey:
+			require.Equal(t, uint16(8), kafkaKey.RequestVersion)
+			numberOfProduceRequests += kafkaStat.Count
+			break
+		case kafka.FetchAPIKey:
+			require.Equal(t, uint16(11), kafkaKey.RequestVersion)
+			numberOfFetchRequests += kafkaStat.Count
+			break
+		default:
+			require.FailNow(t, "Expecting only produce or fetch kafka requests")
+		}
+	}
```

Comment:
add a jira titcket to modify the test so we'll run different produce and fetch versions

Reply:
https://datadoghq.atlassian.net/browse/USMO-262